python/ray/data/dataset.py

```diff
@@ -3628,6 +3630,47 @@ def _aggregate_result(self, result: Union[Tuple, TableRow]) -> U:
         else:
             return result
 
+    def _repr_html_(self) -> str:
+        try:
+            from tabulate import tabulate
+        except ImportError:
+            logger.info(
+                "For rich Dataset reprs in notebooks, " "run `pip install tabulate`."
+            )
+            return ""
+
+        metadata = {
+            "num_blocks": self._plan.initial_num_blocks(),
+            "num_rows": self._meta_count(),
+        }
+
+        schema = self.schema()
+        if schema is None:
+            schema_repr = "<h5>Unknown schema</h5>"
+        elif isinstance(schema, type):
+            schema_repr = f"<h5>Data type: {html.escape(str(schema))}</h5>"
+        else:
+            schema_data = {}
+            for sname, stype in zip(schema.names, schema.types):
+                schema_data[sname] = getattr(stype, "__name__", str(stype))
+
+            schema_repr = tabulate(
+                tabular_data=schema_data.items(),
+                tablefmt="html",
+                showindex=False,
+                headers=["Name", "Type"],
+            )
+
+        return Template("dataset.html.j2").render(
+            schema=schema_repr,
+            metadata=tabulate(
+                tabular_data=metadata.items(),
+                tablefmt="html",
+                showindex=False,
+                headers=["Field", "Value"],
+            ),
```

Comment:
Nit: could we have tabs for "Metadata" and "Schema", with the "Schema" tab first?

Reply:
Yep, I like this change.